scripts/hud/synchronizer.ts

```diff
@@ -0,0 +1,320 @@
+
+const Colors = {
+	EXTRA: ['rgba(24, 150, 211, 1)', 'rgba(87, 200, 255, 1)'],
+	PERFECT: ['rgba(87, 200, 255, 1)', 'rgba(113, 240, 255, 1)'],
+	GOOD: ['rgba(21, 152, 86, 1)', 'rgba(122, 238, 122, 1)'],
+	SLOW: ['rgba(248, 222, 74, 1)', 'rgba(255, 238, 0, 1)'],
+	NEUTRAL: ['rgba(178, 178, 178, 1)', 'rgba(255, 255, 255, 1)'],
+	LOSS: ['rgba(220, 116, 13, 1)', 'rgba(255, 188, 0, 1)'],
+	STOP: ['rgba(211, 24, 24, 1)', 'rgba(255, 87, 87, 1)']
+} satisfies Record<string, [color, color]>;
+
+const DEFAULT_BUFFER_LENGTH = 10;
+const DEFAULT_MIN_SPEED = 200;
+const DEFAULT_SETTING_ON = 1;
+const DEFAULT_SETTING_OFF = 0;
+
+const RAD2DEG = 180 / Math.PI;
+const DEG2RAD = 1 / RAD2DEG;
+
+enum DisplayMode {
+	DISABLED,
+	HALF_WIDTH_THROTTLE,
+	FULL_WIDTH_THROTTLE,
+	STRAFE_INDICATOR,
+	SYNCHRONIZER
+}
+
+enum StatMode {
+	OFF,
+	ON,
+	HIDE_BAR
+}
+
+@PanelHandler()
+class Synchronizer {
+	readonly panels = {
+		wrapper: $('#BarWrapper'),
+		segments: [$('#Segment0'), $('#Segment1'), $('#Segment2'), $('#Segment3'), $('#Segment4')],
+		container: $('#Container'),
+		needle: $('#Needle'),
+		stats: [$<Label>('#StatsUpper'), $<Label>('#StatsLower')]
+	};
+
+	indicatorPercentage = 90; // this value shows ~90% gain or better when strafe indicator touches needle
+	isFirstPanelColored = true; // gets toggled in wrapValueToRange()
+	maxSegmentWidth = 25; // percentage of total element width
+	firstPanelWidth = this.maxSegmentWidth;
+	syncGain = 10; // scale how fast the bars move
+	altColor = 'rgba(0, 0, 0, 0.0)';
+	statMode: StatMode;
+	displayMode: DisplayMode;
+	colorEnable: boolean;
+	dynamicEnable: boolean;
+	statColorEnable: boolean;
+	flipEnable: boolean;
+	interpFrames: number;
+	minSpeed: number;
+	sampleWeight: number;
+	gainRatioHistory: number[];
+	yawRatioHistory: number[];
+
+	initializeSettings() {
+		this.setDisplayMode(GameInterfaceAPI.GetSettingInt('mom_hud_synchro_mode') as DisplayMode);
+		this.setColorMode(GameInterfaceAPI.GetSettingBool('mom_hud_synchro_color_enable'));
+		this.setDynamicMode(GameInterfaceAPI.GetSettingBool('mom_hud_synchro_dynamic_enable'));
+		this.setDirection(GameInterfaceAPI.GetSettingBool('mom_hud_synchro_flip_enable'));
+		this.setBufferLength(GameInterfaceAPI.GetSettingFloat('mom_hud_synchro_buffer_size'));
+		this.setMinSpeed(GameInterfaceAPI.GetSettingFloat('mom_hud_synchro_min_speed'));
+		this.setStatMode(GameInterfaceAPI.GetSettingInt('mom_hud_synchro_stat_mode'));
+		this.setStatColorMode(GameInterfaceAPI.GetSettingBool('mom_hud_synchro_stat_color_enable'));
+	}
+
+	constructor() {
+		RegisterHUDPanelForGamemode({
+			gamemodes: [Gamemode.BHOP, Gamemode.SURF, Gamemode.CLIMB_KZT, Gamemode.CLIMB_MOM],
+			onLoad: () => this.onLoad(),
+			events: [
+				{ event: 'HudProcessInput', callback: () => this.onUpdate() },
+				{ event: 'OnJumpStarted', callback: () => this.onJump() }
+			]
+		});
+
+		$.RegisterForUnhandledEvent('OnSynchroModeChanged', (cvarValue) => this.setDisplayMode(cvarValue));
```

Comment:
same question about registering events for game mode with the panel

Reply:
Yeah same response, if config is always loaded on map init we can pull all events into the famcy thing, will do that.